tests/unit/test_protocol_errors.py

```diff
@@ -22,6 +22,7 @@ class TestFrontendErrors(unittest.TestCase):
     """
     @classmethod
     def setUpClass(cls):
+        reload(frontend)
```

Comment:
True. Best left as another issue. We don't want this PR to be a complete refactor of everything, and this would be something to tackle independently.